griptape/config/base_structure_config.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+from abc import ABC
+from attr import define, field
+from griptape.config import StructureTaskMemoryConfig
+from griptape.drivers import BasePromptDriver
+from griptape.mixins import SerializableMixin
+from griptape.schemas.base_schema import BaseSchema
+from marshmallow import class_registry
+from marshmallow.exceptions import RegistryError
+
+
+@define
+class BaseStructureConfig(SerializableMixin, ABC):
+    prompt_driver: BasePromptDriver = field(kw_only=True, metadata={"save": True})
```

Comment:
Definitely need a better name than `save` but this new metadata field will be used when determining what fields to serialize.

Reply:
since it's going both directions, maybe `serialize` or `serializable`

Reply:
Unity has a cute little pair of decorators for C#: IIRC, they use `[Serializable]` if you want an entire struct/class serialized, or you can make it laser-precise with `[SerializeField]'.

Reply:
Yeah there are a few different syntaxes we could play with here.